RetailCoder.VBE/Inspections/VariableRequiresSetAssignmentEvaluator.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Inspections
+{
+    public static class VariableRequiresSetAssignmentEvaluator
+    {
+        public static IEnumerable<Declaration> GetDeclarationsPotentiallyRequiringSetAssignment(IEnumerable<Declaration> declarations)
+        {
+            //Reduce most of the declaration list with the easy ones
+            var relevantDeclarations = declarations.Where(dec => dec.AsTypeName == Tokens.Variant
```

Comment:
This should be fine if you only call it with non-built-in declarations (`state.AllUserDeclarations`). Why don't you just pass the state in so you can guarantee this?